jax/_src/lax/lax.py

```diff
@@ -1188,7 +1188,7 @@ def squeeze(array: Array, dimensions: Sequence[int]) -> Array:
 
 def expand_dims(array: Array, dimensions: Sequence[int]) -> Array:
   """Insert any number of size 1 dimensions into an array."""
-  ndim_out = np.ndim(array) + len(dimensions)
+  ndim_out = np.ndim(array) + len(set(dimensions))
```

Comment:
Numpy's version of this errors on repeated dimensions. Should we do the same?
```python
>>> x = np.arange(4)
>>> np.expand_dims(x, [1, 1, 1])                                                                                                                                                              
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-3facbd77e287> in <module>
----> 1 np.expand_dims(x, [1, 1, 1])

~/.local/share/virtualenvs/jax-LBbfM5ix/lib/python3.8/site-packages/numpy/core/overrides.py in expand_dims(*args, **kwargs)

~/.local/share/virtualenvs/jax-LBbfM5ix/lib/python3.8/site-packages/numpy/lib/shape_base.py in expand_dims(a, axis)
    595 
    596     out_ndim = len(axis) + a.ndim
--> 597     axis = normalize_axis_tuple(axis, out_ndim)
    598 
    599     shape_it = iter(a.shape)

~/.local/share/virtualenvs/jax-LBbfM5ix/lib/python3.8/site-packages/numpy/core/numeric.py in normalize_axis_tuple(axis, ndim, argname, allow_duplicate)
   1394             raise ValueError('repeated axis in `{}` argument'.format(argname))
   1395         else:
-> 1396             raise ValueError('repeated axis')
   1397     return axis
   1398 

ValueError: repeated axis
```

Reply:
Good observation. Let's try this is in a separate PR, in case it introduces new errors downstream?

Reply:
Sounds good

Reply:
Actually, I think it might be better to do it as part of this PR. It currently results in a less understandable error (because the output shape does not match), and if I'm reading it right, this change makes it so that it's no longer an error (by changing how the output shape is computed).

Reply:
Oops, missed this before merge. Would you like to follow up or should I?

Reply:
Done in #9615